worlds/hades/Locations.py

```diff
@@ -0,0 +1,442 @@
+        **table_weaponlocation,
+        **location_keepsakes,
+        **location_weapons,
+        **location_store_gemstones,
+        **location_store_diamonds,
+        **location_table_fates,
+        **location_table_fates_events,
+    }
+
+def clear_tables():
+    global location_table_tartarus 
+    location_table_tartarus = {
+        "Beat Meg": None,
+    }
+
+    global location_table_asphodel
+    location_table_asphodel = {
+        "Beat Lernie": None,
+    }
+
+    global location_table_elyseum
+    location_table_elyseum = {
+        "Beat Bros": None,
+    }
+
+    global location_table_styx
+    location_table_styx = {
+        "Beat Hades": None,
+    }
+    
+    global location_table_styx_late
+    location_table_styx_late = {
+    }
+    
+
+#Change parameters so they include the settings of the player
+#Chose between old and new system. And for the new system we want to be able
+#to choose how many "locations" we have.
+def setup_location_table_with_settings(options):
+    clear_tables()
+    total_table = {}
+ 
+    total_table.update(location_table_fates_events)
+    
+    if (options.keepsakesanity.value == 1):
+        total_table.update(location_keepsakes)
+     
+    if (options.weaponsanity.value == 1):
+        for weaponLocation, weaponData in location_weapons.items():
+            if (not should_ignore_weapon_location(weaponLocation, options)):
+                total_table.update({weaponLocation : weaponData})
+                
+    if (options.storesanity.value==1):
+        total_table.update(location_store_gemstones)
+        total_table.update(location_store_diamonds)
+
+    if (options.location_system.value==1):
+        result = give_default_location_table()
+        total_table.update(result)
+    elif (options.location_system.value==2):
+        levels = options.score_rewards_amount.value
+        total_table.update(give_score_location_table(levels))
+    elif (options.location_system.value==3):
+        total_table.update(give_weapon_based_locations())
+    
+    if (options.fatesanity==1):
+        total_table.update(location_table_fates)
+    
+    return total_table
+            
+#-----------------------------------------------
+
+def should_ignore_weapon_location(weaponLocation, options):
+    if (options.initial_weapon.value == 0 and weaponLocation == "SwordWeaponUnlockLocation"):
+        return True
+    if (options.initial_weapon.value == 1 and weaponLocation == "BowWeaponUnlockLocation"):
+        return True
+    if (options.initial_weapon.value == 2 and weaponLocation == "SpearWeaponUnlockLocation"):
+        return True
+    if (options.initial_weapon.value == 3 and weaponLocation == "ShieldWeaponUnlockLocation"):
+        return True
+    if (options.initial_weapon.value == 4 and weaponLocation == "FistWeaponUnlockLocation"):
+        return True
+    if (options.initial_weapon.value == 5 and weaponLocation == "GunWeaponUnlockLocation"):
+        return True
+    return False;
```

Comment:
```suggestion
    return False
```
Also the rest of the function could use some work but whatever, it's fine